floodlight/io/dfl.py

```diff
@@ -342,24 +281,115 @@ def read_pitch_from_mat_info_xml(filepath_mat_info: Union[str, Path]) -> Pitch:
     return pitch
 
 
-def read_event_data_xml(
-    filepath_events: Union[str, Path]
-) -> Tuple[Events, Events, Events, Events]:
-    """Parses a DFL Match Event XML file and extracts the event data.
+def read_teamsheets_from_mat_info_xml(filepath_mat_info) -> Dict[str, Teamsheet]:
+    """Reads match_information XML file and returns two teamsheet objects for the home
+    and the away team.
+
+    Parameters
+    ----------
+    filepath_mat_info: str or pathlib.Path
+        Full path to XML File where the Match Information data in DFL format is saved.
+
+    Returns
+    -------
+    teamsheets: Dict[str, Teamsheet]
+        Dictionary with teamsheets for the home team and the away team.
+    """
+    # set up XML tree
+    tree = etree.parse(str(filepath_mat_info))
+    root = tree.getroot()
+
+    # initialize teamsheets
+    teamsheets = {
+        "Home": pd.DataFrame(
+            columns=["player", "position", "team", "jID", "pID", "tID"]
+        ),
+        "Away": pd.DataFrame(
+            columns=["player", "position", "team", "jID", "pID", "tID"]
+        ),
+    }
+
+    # find team ids
+    team_informations = root.find("MatchInformation").find("Teams")
+    home_id = root.find("MatchInformation").find("General").get("HomeTeamId")
+    if "AwayTeamId" in root.find("MatchInformation").find("General").attrib:
+        away_id = root.find("MatchInformation").find("General").get("AwayTeamId")
+    elif "GuestTeamId" in root.find("MatchInformation").find("General").attrib:
+        away_id = root.find("MatchInformation").find("General").get("GuestTeamId")
+    else:
+        away_id = None
+
+    # parse player information
+    for team_info in team_informations:
+        if team_info.get("TeamId") == home_id:
+            team = "Home"
+        elif team_info.get("TeamId") == away_id:
+            team = "Away"
+        else:
+            team = None
+
+        # skip referees sometimes referred to as a team in new data formats
+        if team not in ["Home", "Away"]:
+            continue
 
-    This function provides a high-level access to the particular DFL Match Event feed
-    and returns Event objects for both teams. The number of segments is inferred from
-    the data, yet data for each segment is stored in a separate object.
+        # create list of players
+        players = team_info.find("Players")
+
+        # create teamsheets
+        teamsheets[team]["player"] = [player.get("Shortname") for player in players]
+        teamsheets[team]["pID"] = [player.get("PersonId") for player in players]
+        teamsheets[team]["jID"] = [int(player.get("ShirtNumber")) for player in players]
```

Comment:
you could use the ``get_and_convert()`` method here, in the rare case there is no ShirtNumber in the data and the ``get()`` returns None

Reply:
Agreed, included for all player attributes now.